### PR TITLE
Travis python37

### PR DESCRIPTION
**Pull request by dcbaker**
Test alot on python 3.7 in CI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,15 +7,15 @@ cache:
   ccache: True
 
 
-# We need a newer version of zlib to build notmuch from source as is available
-# in precise.
-dist: trusty
+# Python 3.7 is not supported in trusty
+dist: xenial
```
Comment (lucc) [inline .travis.yml]: Since when is that possible? I never saw it in the docs? (But the log claims it works.)

Comment (dcbaker) [reply .travis.yml]: I found it in a bug report about python 3.7 support. It's apparently not fully supported yet, but it seems to work. The suggested workaround for python 3.7 support was to build a docker image, which we can do but it's much more involved than just setting "xenial"

Comment (lucc) [reply .travis.yml]: Can we live with some older libraries than the ones we currently build by hand? Here is what we build by hand and what xenial has:
* gpgme 1.9 -> [1.6](https://packages.ubuntu.com/xenial/libgpgme11)
* libassuan (dep for gpgme) 2.4.3 -> [2.4.2](https://packages.ubuntu.com/xenial/libassuan-dev)
* libgpg-error (dep for gpgme) 1.27 -> [1.21](https://packages.ubuntu.com/xenial/libgpg-error-dev)
* notmuch (we require 0.23 in setup.py) git (xenial is to low but we could switch to a tarball)

Comment (dcbaker) [reply .travis.yml]: I think it's worth trying to use the existing versions of stuff (I'm pretty sure that libassuan can be dropped at least). I'll push a few patches and see what happens.

File: .travis.yml [hunk 2]
```diff
@@ -35,6 +35,7 @@ addons:
       - libtalloc-dev
       - zlib1g-dev
       - libgmime-2.6-dev
+      - libassuan-dev
```
Comment (lucc) [inline .travis.yml]: Is it a dependency of notmuch or gpg?  Se the comment above.,

Comment (dcbaker) [reply .travis.yml]: Doesn't seem happy with libgpgme 1.6: https://travis-ci.org/pazz/alot/jobs/411043268

**Comment by pazz** [comment]
https://travis-ci.org/pazz/alot/jobs/411183305
is this related?

**Comment by lucc** [comment]
No, after restarting it it completes: https://travis-ci.org/pazz/alot/jobs/411183305

**Comment by pazz** [comment]
strange, but ok :) cheers!

**Comment by pazz** [comment]
by the way: should we also mention the new python version in our setup.py?

**Comment by dcbaker** [comment]
ah, yes we should. I'll make a new PR

**Pull request merged**